server/valgrind-out.txt

```diff
@@ -0,0 +1,128 @@
+--8994-- REDIR: 0x48fb820 (libc.so.6:strpbrk) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fb100 (libc.so.6:index) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fb630 (libc.so.6:strlen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x4904bb0 (libc.so.6:memrchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fc850 (libc.so.6:strcasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fc3d0 (libc.so.6:memchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x4918930 (libc.so.6:wcslen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fbae0 (libc.so.6:strspn) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fc750 (libc.so.6:stpncpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fc6f0 (libc.so.6:stpcpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fdc90 (libc.so.6:strchrnul) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x48fc8a0 (libc.so.6:strncasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--8994-- REDIR: 0x49e4500 (libc.so.6:__strrchr_avx2) redirected to 0x483ea10 (rindex)
+--8994-- REDIR: 0x49e7b60 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x48428e0 (memset)
+--8994-- REDIR: 0x48f60e0 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
+--8994-- REDIR: 0x48f66d0 (libc.so.6:free) redirected to 0x483c9d0 (free)
+--8994-- REDIR: 0x49e4310 (libc.so.6:__strchrnul_avx2) redirected to 0x4843540 (strchrnul)
+--8994-- REDIR: 0x49e5c10 (libc.so.6:__strcpy_avx2) redirected to 0x483f090 (strcpy)
+--8994-- REDIR: 0x48f7b10 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
+--8994-- REDIR: 0x49e46d0 (libc.so.6:__strlen_avx2) redirected to 0x483ef40 (strlen)
+--8994-- REDIR: 0x49e76e0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x48429f0 (memmove)
+--8994-- REDIR: 0x48fbfa0 (libc.so.6:__GI_strstr) redirected to 0x4843ca0 (__strstr_sse2)
+--8994-- REDIR: 0x49e0cc0 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x48421e0 (bcmp)
+--8994-- REDIR: 0x49dfbd0 (libc.so.6:__strcmp_avx2) redirected to 0x483fed0 (strcmp)
+--8994-- REDIR: 0x49e76c0 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4843660 (mempcpy)
+--8994-- REDIR: 0x48f6e80 (libc.so.6:realloc) redirected to 0x483df30 (realloc)
+==8994== 
+==8994== HEAP SUMMARY:
+==8994==     in use at exit: 64 bytes in 1 blocks
+==8994==   total heap usage: 317 allocs, 316 frees, 2,328,361 bytes allocated
+==8994== 
+==8994== Searching for pointers to 1 not-freed blocks
+==8994== Checked 71,648 bytes
+==8994== 
+==8994== 64 bytes in 1 blocks are still reachable in loss record 1 of 1
+==8994==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==8994==    by 0x495FBF9: gaih_inet.constprop.0 (getaddrinfo.c:1058)
+==8994==    by 0x4961F58: getaddrinfo (getaddrinfo.c:2256)
```

Comment:
@Nileshkartik95 this is highlighting that you are missing cleanup from `getaddrinfo`